### ISSUE TITLE
Move recycling of GpuImages to a lower abstraction layer

### ISSUE THREAD

**Issue by emilwestergren** [cogneco/ooc-kean#1067]
Currently `OpenGLPacked` images (packed `GpuImages`) are recycled in the `GpuContext` they are created in. I suggest we move this recycling to a lower level (OpenGL texture level). This is more in sync with the recycling of `ByteBuffer` for `RasterImages`.

* The cost is some extra memory allocations every frame
* Will remove the need to modify private members like `_coordinateSystem` when recycling images
* Hides recycling from the "user layer"

@sebastianbaginski @simonmika opinions?

**Comment by sebastianbaginski** [comment]
> The cost is some extra memory allocations every frame

How much will it cost in terms of processing time per frame ?
> Hides recycling from the "user layer"

I think it's a good idea.

**Comment by simonmika** [comment]
Go ahead. It is a good idea as we most likely will keep the luma part around for longer then the color.

**Comment by emilwestergren** [comment]
> How much will it cost in terms of processing time per frame ?
Not that much, it's just the extra allocation and deallocation of object data for the high level images.

**Comment by marcusnaslund** [comment]
Is this issue still relevant? @emilwestergren

**Comment by emilwestergren** [comment]
Yes, but it's not urgent and it might not be possible with our current architecture.
